src/blockdb/blockdb.h

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2018 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BLOCKDB_H
+#define BLOCKDB_H
+
+#include "uint256.h"
+#include "dbwrapper.h"
+#include "chain.h"
+#include "primitives/block.h"
+#include "undo.h"
+
+struct BlockDBValue
+{
+    int32_t blockVersion;
+    uint64_t blockHeight;
+    CBlock block;
+
+    BlockDBValue()
+    {
+        SetNull();
+    }
+
+    BlockDBValue(const CBlock &_block)
+    {
+        assert(_block.IsNull() == false);
+        this->block = _block;
+        this->blockVersion = this->block.nVersion;
```

Comment:
Not sure I see the need for a BlockDBValue.  I see that you're trying to store blockheight and version but that data is already in the blockindex?  The main problem though I see here is that to create this blockdbvalue you have to copy the block each time to create it.

Reply:
it was outlined in the original spec zerg wrote. https://bitco.in/forum/threads/devpool-work-replace-block-files-with-key-value-db.8532/